akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -162,23 +162,20 @@ private[akka] class AffinityPool(
         indexFor(runnableHash)
       else {
         @tailrec
-        def updateAndOrGetIndex(): Int = {
-          val prev = hashCache.get()
-          if (prev.size > fairDistributionThreshold) indexFor(runnableHash)
```

Comment:
I was expecting a bit more of a boost to be honest. I guess the hashcode generation is really not all that expensive in the bigger picture of things. However, indeed no need to generate them before checking the hash cache

![fixed-redunant-hascode-computation](https://user-images.githubusercontent.com/4391506/28386971-2c916366-6c82-11e7-8a36-59d4d65e1600.png)